src/platform/common/configMigration.ts

```diff
@@ -30,45 +30,61 @@ export class ConfigMigration {
 
     constructor(private readonly jupyterConfig: WorkspaceConfiguration) {}
 
-    public async migrateSettings() {
+    public migrateSettings() {
         for (let prop of Object.keys(ConfigMigration.migratedSettings)) {
-            await this.migrateSetting(prop, ConfigMigration.migratedSettings[prop]);
+            this.migrateSetting(prop, ConfigMigration.migratedSettings[prop]);
         }
```

Comment:
How about something like `await Promise.all(ConfigMigration.migratedSettings.values())` instead (although `Promise.allSettled` is better for handling errors)

Reply:
btw, if you make this change, then you can change `migrateSetting` to use `await` again